src/Grid/EventButton.vala

```diff
@@ -55,17 +57,26 @@ public class Maya.View.EventButton : Gtk.Revealer {
                 Gtk.Menu menu = new Gtk.Menu ();
                 menu.attach_to_widget (this, null);
 
-                var edit_item = new Gtk.MenuItem.with_label (_("Edit…"));
-                var remove_item = new Gtk.MenuItem.with_label (_("Remove"));
+                bool sensitive = src.writable == true && Model.CalendarModel.get_default ().calclient_is_readonly (src) == false;
 
-                if (src.writable != true && Model.CalendarModel.get_default ().calclient_is_readonly (src) != false) {
-                    edit_item.sensitive = false;
-                    remove_item.sensitive = false;
+                Gtk.MenuItem edit_item = new Gtk.MenuItem.with_label (_("Edit…"));
+                edit_item.activate.connect (() => { edition_request (); });
+                edit_item.sensitive = sensitive;
+                menu.append (edit_item);
+
+                Gtk.MenuItem remove_item;
+                if (comp.has_recurrences ()) {
+                    remove_item = new Gtk.MenuItem.with_label (_("Remove series"));
+                    Gtk.MenuItem exception_item = new Gtk.MenuItem.with_label (_("Remove occurrence"));
+                    exception_item.activate.connect (() => { add_exception (); });
```

Comment:
This could be written more simply as `exception_item.activate.connect (add_exception);`